clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -303,6 +300,7 @@ protected void onJoinComplete(int generation,
      */
     public boolean poll(final long timeoutMs) {
         final long startTime = time.milliseconds();
+        long currentTime = startTime;
```

Comment:
Do you need this currentTime because `ensureCoordinatorReady`, `ensureFreshMetadata` and `ensureActiveGroup` are somewhat long-running?

Reply:
Yes, right. I think the approach in #5087 is less irritating.